friends_3ds.go

```diff
@@ -83,6 +111,28 @@ func NewMii() *Mii {
 	return &Mii{}
 }
 
+type FriendMii struct {
+	PrincipalID uint32
```

Comment:
This applies to the whole codebase but I won't make review comments everywhere this happens

The real names for these fields is unknown, and I am aware that Kinnay's wiki names them as "principal ID". This is correct, however to maintain consistency with the rest of the codebase this should probably be renamed to `PID` instead of `PrincipalID`, unless the real name is known later on